File: output/gemini_3_1_pro_preview/zero/cpp_noimg/Fita Colorida.cpp

#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int n;
    if (!(cin >> n)) return 0;

    vector<int> fita(n);
    for (int i = 0; i < n; ++i) {
        cin >> fita[i];
    }

    vector<int> dist(n, 1e9);
    int last_zero = -1;

    for (int i = 0; i < n; ++i) {
        if (fita[i] == 0) {
            dist[i] = 0;
            last_zero = i;
        } else if (last_zero != -1) {
            dist[i] = i - last_zero;
        }
    }

    last_zero = -1;
    for (int i = n - 1; i >= 0; --i) {
        if (fita[i] == 0) {
            dist[i] = 0;
            last_zero = i;
        } else if (last_zero != -1) {
            dist[i] = min(dist[i], last_zero - i);
        }
    }

    for (int i = 0; i < n; ++i) {
        cout << min(9, dist[i]) << (i == n - 1 ? "" : " ");
    }
    cout << "\n";

    return 0;
}

File: output/csbc_semish_2026/deepseek_deepseek_v3_2/few/python/Fita Colorida.py
import sys

def main():
    data = sys.stdin.read().strip().split()
    if not data:
        return
    n = int(data[0])
    fita = list(map(int, data[1:1+n]))
    
    INF = 10**9
    dist_left = [INF] * n
    dist_right = [INF] * n
    
    # da esquerda para direita
    last_zero = -INF
    for i in range(n):
        if fita[i] == 0:
            last_zero = i
        dist_left[i] = i - last_zero
    
    # da direita para esquerda
    last_zero = INF
    for i in range(n-1, -1, -1):
        if fita[i] == 0:
            last_zero = i
        dist_right[i] = last_zero - i
    
    # resultado: menor distância, limitada a 9
    res = []
    for i in range(n):
        d = min(dist_left[i], dist_right[i])
        if d > 9:
            d = 9
        res.append(str(d))
    
    print(" ".join(res))

if __name__ == "__main__":
    main()